Models/ECommerceCurrencySiteSettingsPart.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Orchard.ContentManagement;
+using Orchard.Environment.Extensions;
+
+namespace Nwazet.Commerce.Models {
+    [OrchardFeature("Nwazet.CurrencyProviderBySiteSetting")]
+    public class ECommerceCurrencySiteSettingsPart : ContentPart {
+        [Required,MaxLength(3)]
+        public string CurrencyCode
+        {
+            get {
+                string cc = this.Retrieve(p => p.CurrencyCode);
+                return string.IsNullOrWhiteSpace(cc) ? "USD" : cc; //default is USD
```

Comment:
I feel that using the site's culture is a rather clunky way of initializing this site's setting with a default value. Also a `System.Globalization.NumberFormatInfo` object doe not seem to contain a currency's 3-letters code, but only its symbol.